### PR TITLE
Make ESXi Plugin work without crypto and fix vm_inventory

### PR DESCRIPTION
**Pull request by Matthijsy**
Currently the ESXi module works partly without having PyCryptodome installed, however this package is not 100% needed. Furthermore the vm_inventory method did always return the vm list. This PR does 3 things:

- Check if PyCryptodome or Pystandalone is availble, if not do not use the envelope
- Don't crash if there is no ESXi config, this makes the vm_inventory method available without crypto
- Creating the local layer after the mount modules

Currently the local layer is created before the mount modules. Some of these modules (in my case vim.v00) can contain empty `vmInventory.xml` files, which results in the `vm_inventory` method returning an empty list. By creating this local layer later the version in the local layer is used.

### CODE REVIEW

**Comment by Matthijsy** [comment]
@DissectBot agree company="Northwave"

File: dissect/target/plugins/os/unix/esxi/_os.py [hunk 1]
```diff
@@ -37,6 +38,7 @@
     ],
 )
 
+log = logging.getLogger(__name__)
```
Comment (Schamper) [inline dissect/target/plugins/os/unix/esxi/_os.py:41]:
```suggestion
```

File: dissect/target/plugins/os/unix/esxi/_os.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 import gzip
 import json
+import logging
```
Comment (Schamper) [inline dissect/target/plugins/os/unix/esxi/_os.py:5]:
```suggestion
```

File: dissect/target/plugins/os/unix/esxi/_os.py [hunk 3]
```diff
@@ -70,7 +72,8 @@ def __init__(self, target: Target):
 
     def _cfg(self, path: str) -> Optional[str]:
         if not self._config:
-            raise ValueError("No ESXi config!")
+            log.warning("No ESXi config!")
```
Comment (Schamper) [inline dissect/target/plugins/os/unix/esxi/_os.py:75]:
```suggestion
            self.target.log.warning("No ESXi config!")
```
We can use the target-bound logging to get additional context on the target where the warning occurred.

**Review by Schamper — CHANGES_REQUESTED** [review]

**Review by Schamper — APPROVED** [review]

**Review by Schamper — APPROVED** [review]

**Pull request merged**